server/routes/tenants.js

```diff
@@ -11,19 +11,61 @@ const router = express.Router();
 router.get('/tenants/:tenantId', async (req, res, next) => {
   const tenantId = req.params.tenantId;
   try {
-    const foundTenant = await Tenant.findById(tenantId).populate('propertyId');
```

Comment:
@zczhou16 I don't think really think these changes are necessary. Moreover, it slows down the endpoint since DB processes are fairly slow.  @ryanzhxu and I discussed previously that a tenant must belong to a property so such a change is redundant. As I mentioned, for the purpose of the demo we populated the tenants collection with a bunch of dummy data, and in some cases failed to ensure that this specific requirement was met, which is why you may have been running into issues. If a tenant does not belong to an existing property, it should not exist in the DB in the first place.